src/test/java/com/varukha/onlinebookstore/controller/CategoryControllerTest.java

```diff
@@ -0,0 +1,244 @@
+        mockMvc = MockMvcBuilders
+                .webAppContextSetup(applicationContext)
+                .apply(springSecurity())
+                .build();
+        teardown(dataSource);
+        try (Connection connection = dataSource.getConnection()) {
+            connection.setAutoCommit(true);
+            ScriptUtils.executeSqlScript(
+                    connection,
+                    new ClassPathResource(
+                            "database/category/add-category-to-category-table.sql")
+            );
+        }
+    }
+
+    @AfterAll
+    static void afterAll(
+            @Autowired DataSource dataSource
+    ) {
+        teardown(dataSource);
+    }
+
+    @SneakyThrows
+    static void teardown(DataSource dataSource) {
+        try (Connection connection = dataSource.getConnection()) {
+            connection.setAutoCommit(true);
+            ScriptUtils.executeSqlScript(
+                    connection,
+                    new ClassPathResource(
+                            "database/category/remove-category-from-category-table.sql")
+            );
+        }
+    }
+
+    @Test
+    @WithMockUser(username = "admin", roles = {"ADMIN"})
+    @DisplayName("""
+            Test the 'save' new category endpoint
+            with valid request parameters
+             """)
+    void save_ValidCategoryDtoRequestData_ReturnCategoryDto() throws Exception {
+        CreateCategoryRequestDto categoryRequestDto = new CreateCategoryRequestDto();
+        categoryRequestDto.setName("Historical Fiction");
+        categoryRequestDto.setDescription("Historical Fiction category description");
+
+        CategoryDto categoryDtoResponse = new CategoryDto();
+        categoryDtoResponse.setId(4L);
+        categoryDtoResponse.setName(categoryRequestDto.getName());
+        categoryDtoResponse.setDescription(categoryRequestDto.getDescription());
+
+        String jsonRequest = objectMapper.writeValueAsString(categoryRequestDto);
+        MvcResult result = mockMvc.perform(post("/categories")
+                        .content(jsonRequest)
+                        .contentType(MediaType.APPLICATION_JSON))
+                .andExpect(status().isCreated())
+                .andReturn();
+        CategoryDto actualCategoryDto = objectMapper.readValue(result.getResponse()
+                .getContentAsString(), CategoryDto.class);
+        assertNotNull(actualCategoryDto);
+        assertNotNull(actualCategoryDto.getId());
+        EqualsBuilder.reflectionEquals(categoryDtoResponse, actualCategoryDto);
+    }
+
+    @Test
+    @WithMockUser(roles = {"ADMIN", "USER"})
+    @DisplayName("""
+            Test the 'getAll' endpoint with valid request parameters
+            and pagination to retrieve all book categories.
+            """)
+    void getAll_ValidCategoryDtoRequestData_ReturnCategoryDto() throws Exception {
+        Category categoryNovel = new Category();
```

Comment:
Please make some constants to not create preconditions in each test or create some methods or even class to make preconditions (like CategoryObjectMother with methods to create categories)